src/pages/home/index.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { useEffect, useState } from "react";
+import { decode } from "../../utils/jwt";
+import "./style.css";
+
+interface UserTypes {
+  data: {
+    userId: string;
+  };
+}
+
+const getUserId = () => {
+  const hasUser = window.localStorage.getItem("auth-token");
+  if (hasUser) {
+    const userId: Partial<UserTypes> = decode(hasUser);
+    return userId;
+  }
+};
+
+export const HomePage = () => {
+  const [userId, setUserId] = useState<Partial<UserTypes>>();
+
+  useEffect(() => {
+    setUserId(getUserId());
+  }, []);
```

Comment:
aqui poderia usar lazy initialization do `useState`, assim a `getUserId` seria chamada apenas uma vez e não precisaria do `useEffect`
```suggestion
  const [userId, setUserId] = useState<Partial<UserTypes>>(() => getUserId());
```

Reply:
Removi a parte de trazer o id de usuário para a página, mas vou guardar a sugestão para futuras implementações. Obrigado Rafael!